src/D2L.CodeStyle.Analyzers/Language/AttributeUnqualifiedNameAnalyzer.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System.Collections.Immutable;
+using D2L.CodeStyle.Analyzers.Extensions;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace D2L.CodeStyle.Analyzers.Language {
+
+	[DiagnosticAnalyzer( LanguageNames.CSharp )]
+	internal sealed class AttributeUnqualifiedNameAnalyzer : DiagnosticAnalyzer {
+
+		public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(
+			Diagnostics.ConciseAttributeName
+		);
+
+		public override void Initialize( AnalysisContext context ) {
+			context.EnableConcurrentExecution();
+			context.ConfigureGeneratedCodeAnalysis( GeneratedCodeAnalysisFlags.Analyze | GeneratedCodeAnalysisFlags.ReportDiagnostics );
+			context.RegisterCompilationStartAction( CompilationStart );
+		}
+
+		private void CompilationStart( CompilationStartAnalysisContext context ) {
+
+			context.RegisterSyntaxNodeAction(
+					c => AnalyzeAttribute( c, (AttributeSyntax)c.Node ),
+					SyntaxKind.Attribute
+				);
+		}
+
+		private void AnalyzeAttribute(
+				SyntaxNodeAnalysisContext context,
+				AttributeSyntax attribute
+			) {
+
+			string unqualifiedName = attribute.Name
+				.GetUnqualifiedName()
+				.ToString();
+
+			if( !unqualifiedName.EndsWith( "Attribute" ) ) {
+				return;
+			}

```

Comment:
So `class JsonAttributeAttribute` would be unusable?
I'm not super worried about that or anything.
(silly example because json doesn't have "attributes" but like, I don't know maybe there could be a case like that for something.)

Reply:
basically I'm asking: is it worth the effort to try harder here?

Reply:
So Visual Studio already complains with a Style warning:

![image](https://user-images.githubusercontent.com/12101120/151880443-3e2dd1bc-81e7-4db8-83d0-96d47e3f1879.png)

https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0001

I see no value in being able to specify the attributes 2 different ways.

Reply:
Right - Jacob's explaining a case where "Attribute" is part of the domain model, which this is insensitive to.

```csharp

public class XmlSerializerAttribute : Attribute {}
public class XmlAttributeAttribute : Attribute {}
public class XmlElementAttribute : Attribute {}

[XmlSerializer]
public class Person {

  [XmlAttribute]
  public string FirstName { get; }
  
  [XmlElement]
  public string LastName { get; }

}
```

```xml
<person firstName="John">
  <lastName>Smith</lastName>
</person>
```

This would complain about the correct `[XmlAttribute]` usage.

Reply:
No I get that, but that doesn't exist in the dotnet framework or our framework, so not super concerned.

Reply:
Oh it does exist:

https://source.dot.net/#System.Private.Xml/System/Xml/Serialization/XmlAttributeAttribute.cs,64ee1e30fec3074f,references

Reply:
I guess it doesn't search suffixes:

![image](https://user-images.githubusercontent.com/12101120/151884498-f3057aca-a474-444a-995d-991ef1c8248e.png)